### PR TITLE
Add gate for feature labels

### PR DESCRIPTION
**Pull request by therealmitchconnors**
This PR makes new integration tests which do not declare feature labels fail, pointing them to documentation on how to properly label tests.  The whitelist.txt file lists all known integ test cases today.

### CODE REVIEW

**Comment by therealmitchconnors** [comment]
Fixes #24563

File: pkg/test/framework/test.go [hunk 1]
```diff
@@ -208,6 +208,14 @@ func (t *Test) runInternal(fn func(ctx TestContext), parallel bool) {
 		return
 	}
 
+	// TODO: should we also block new cases?
+	suiteName := t.s.settings.TestID
+	if len(t.featureLabels) < 1 && !features.GlobalWhitelist.Contains(suiteName, t.name) {
+		scopes.Framework.Errorf("Detected new test %s in suite %s with no feature labels.  "+
+			"See istio/istio/pkg/test/framework/features/README.md", suiteName, t.name)
+		t.goTest.FailNow()
```
Comment (howardjohn) [inline pkg/test/framework/test.go:216]: can you do goTest.Fatal("error") the it will be output more clearly I think?

Comment (therealmitchconnors) [reply pkg/test/framework/test.go:216]: Thank you.  I knew there was a function like that, but couldn't find it..

File: pkg/test/framework/test.go [hunk 2]
```diff
@@ -198,6 +198,20 @@ func (t *Test) runInternal(fn func(ctx TestContext), parallel bool) {
...
+	if t.goTest != nil {
+		myGoTest = t.goTest
+	} else {
+		myGoTest = t.parent.goTest
+	}
+	suiteName := t.s.settings.TestID
+	if len(t.featureLabels) < 1 && !features.GlobalWhitelist.Contains(suiteName, myGoTest.Name()) {
+
+		myGoTest.Fatalf("Detected new test %s in suite %s with no feature labels.  "+
+			"See istio/istio/pkg/test/framework/features/README.md", myGoTest.Name(), suiteName)
```
Comment (howardjohn) [inline pkg/test/framework/test.go:212]:
```suggestion
			"See istio/pkg/test/framework/features/README.md", myGoTest.Name(), suiteName)
```

File: pkg/test/framework/features/whitelist.txt [hunk 3]
```diff
@@ -0,0 +1,1679 @@
...
+authz_deny,TestAuthzDeny
+authz_ingress,TestAuthzIngress
+authz_jwt,TestAuthzJWT
+authz_tcp,TestAuthzTCP
+cert_mtls_test,TestCertMtls
+cert_provision_prometheus,TestPrometheusCert
+client_tracing_test,TestClientTracing
+cni,TestCNIReachability
+cni,TestCNIReachability/global-mtls-on
+cni,TestCNIReachability/global-mtls-on/a->grpc://a:grpc
```
Comment (howardjohn) [inline pkg/test/framework/features/whitelist.txt:14]: maybe we should only add the top level? like TestCNIReachability here. just to make it a bit smaller

File: pkg/test/framework/test.go [hunk 4]
```diff
@@ -198,6 +198,20 @@ func (t *Test) runInternal(fn func(ctx TestContext), parallel bool) {
...
 
+	// TODO: should we also block new cases?
+	var myGoTest *testing.T
+	if t.goTest != nil {
+		myGoTest = t.goTest
+	} else {
+		myGoTest = t.parent.goTest
+	}
+	suiteName := t.s.settings.TestID
+	if len(t.featureLabels) < 1 && !features.GlobalWhitelist.Contains(suiteName, myGoTest.Name()) {
```
Comment (howardjohn) [inline pkg/test/framework/test.go:209]: should we warn if there is something in whitelist that *does* have the label? that way we can clean it up?

Probably not fail though

**Pull request merged**